### PR TITLE
5560: [TEST]: Eliminate cleanup sections pt.2 (Ha-Flows)

### PR DESCRIPTION
**Pull request by pkazlenka**
Implements #5560

* Removed cleanup actions for Ha-Flow tests
* Refactored HaFlowExtended class to be based on model classes instead of DTOs

### CODE REVIEW

**Review by yuliiamir — COMMENTED** [review]
Could you please also update Server42HaFlowRttSpec?

**Review by izadorozhna — APPROVED** [review]

File: src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/model/HaFlowExtended.groovy [hunk 1]
```diff
@@ -100,28 +121,70 @@ class HaFlowExtended {
...
+                                    .portNumber(it.endpointPort)
+                                    .vlanId(it.endpointVlan)
+                                    .innerVlanId(it.endpointInnerVlan)
+                                    .build())
+                            .description(it.description)
+                            .build()
+                })
+                .build())
+        waitForBeingInState(expectedState)
+        return new HaFlowExtended(haFlow, northboundV2, topologyDefinition, cleanupManager)
```
Comment (yuliiamir) [inline src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/model/HaFlowExtended.groovy:180]: In the scope of flow creation, we wait for an ha-flow to be in UP state, but method returns HaFlowExtended from northbound response(flow is in_progress state).

**Comment by pkazlenka** [comment]
> Could you please also update Server42HaFlowRttSpec?

Updated s42 and stats specs as well. Thank you for reminding me.

File: src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/model/HaFlowExtended.groovy (hunk 1 above)
Comment (pkazlenka) [reply src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/model/HaFlowExtended.groovy]: Fixed. Now `create()` returns what `waitForBeingInState()` returns.

**Review by yuliiamir — APPROVED** [review]
Could you please delete  an empty file src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/YFlowHelper.groovy ?

**Comment by pkazlenka** [comment]
> Could you please delete an empty file src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/YFlowHelper.groovy ?

It's being deleted by this PR.

**Pull request merged**